src/util/mod.rs

```diff
@@ -327,6 +403,13 @@ pub fn client_mobile(hdr: &Headers) -> bool {
     hdr.get::<UserAgent>().map(|s| s.contains("Mobi") || s.contains("mobi")).unwrap_or(false)
 }
 
+/// Check if, given the request headers, the client should be treated as Microsoft software.
+///
+/// Based on https://github.com/miquels/webdav-handler-rs/blob/02433c1acfccd848a7de26889f6857cbad559076/src/handle_props.rs#L529
+pub fn client_microsoft(hdr: &Headers) -> bool {
+    hdr.get::<UserAgent>().map(|s| s.contains("Microsoft") || s.contains("microsoft")).unwrap_or(false)
```

Comment:
Wouldn't we just want to check for containing 'microsoft' with out caring about case sensitivity at all? ie, allow `MICROSOFT` to match

Reply:
No client in my testing used allcaps for this. Hell, no client I know of uses allcaps for the UA at *all*.